beacon-chain/p2p/handshake.go

```diff
@@ -54,37 +56,47 @@ func (s *Service) AddConnectionHandler(reqFunc func(ctx context.Context, id peer
 						// used in production.
 						log.Debug("Not disconnecting peer with unsupported protocol. This may be the DHT node or relay.")
 						s.host.ConnManager().Protect(conn.RemotePeer(), "relay/bootnode")
+						s.peers.SetConnectionState(conn.RemotePeer(), peers.PeerDisconnected)
```

Comment:
we are still connected with this peer though. Putting its state as disconnected isnt correct

Reply:
This is tracking peers at the protocol level, not the network level.  If it isn't a valid Ethereum 2 peer (i.e. goes through the handshake process) it is not a connected peer.

Reply:
Resolving as per conversation.